pkg/strategy/xgap/strategy.go

```diff
@@ -237,17 +237,28 @@ func (s *Strategy) CrossRun(ctx context.Context, _ bbgo.OrderExecutionRouter, se
 }
 
 func (s *Strategy) makeSpread(ctx context.Context, bestBid, bestAsk types.PriceVolume) error {
-	orderForms := []types.SubmitOrder{
-		{
-			Symbol:      s.Symbol,
-			Side:        types.SideTypeSell,
-			Type:        types.OrderTypeLimit,
-			Quantity:    fixedpoint.Max(bestBid.Volume, s.tradingMarket.MinQuantity),
-			Price:       bestBid.Price,
-			Market:      s.tradingMarket,
-			TimeInForce: types.TimeInForceIOC,
-		},
+	balances := s.tradingSession.GetAccount().Balances()
+	order := types.SubmitOrder{
+		Symbol:      s.Symbol,
+		Side:        types.SideTypeSell,
+		Type:        types.OrderTypeLimit,
+		Quantity:    fixedpoint.Max(bestBid.Volume, s.tradingMarket.MinQuantity),
+		Price:       bestBid.Price,
+		Market:      s.tradingMarket,
+		TimeInForce: types.TimeInForceIOC,
+	}
+	if err := tradingutil.HasSufficientBalance(
+		s.tradingMarket,
+		order,
+		balances,
+	); err != nil {
+		return fmt.Errorf(
+			"cannot make spread for %s: %s",
+			s.tradingMarket.BaseCurrency,
+			err,
+		)
 	}
+	orderForms := []types.SubmitOrder{order}
```

Comment:
we will always have only one make spread order, hence no need to create a slice?

Reply:
I adopt this from the original code.